posts/the-game-of-vue-migrations.md

```diff
@@ -0,0 +1,158 @@
+
+Though the Vue template only calls "`generateNextGenerationGrid`", overall scoping smaller methods in and outside a component is better for a project's longevity. Of course, utilizing TypeScript's [`enum`](https://www.typescriptlang.org/docs/handbook/enums.html) and [`type`/`interface`](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces) features can help too. Consolidating [similar lines of code](https://github.com/matthewreed26/game-of-life/blob/unit-jest-typescript/src/components/game-of-life/GameOfLife.component.ts#L53) is a worthwhile place to start. Extracting these lines into anonymous functions and externalizing them to [separate files](https://github.com/matthewreed26/game-of-life/blob/unit-jest-typescript-intentionality/src/components/game-of-life/NeighborCell.ts) allows [direct testing without going through the Vue Testing Utils](https://github.com/matthewreed26/game-of-life/blob/unit-jest-typescript-intentionality/tests/unit/components/game-of-life/NeighborCell.spec.ts). Not to mention abstracting out code [decouples](<https://en.wikipedia.org/wiki/Coupling_(computer_programming)>) the dependency on the "`this`" context of the Vue component.
+
+## Say Hello to the New Vue
+
+The guidance for upgrading 2 to 3 previously shared highlights some [limitations](https://v3.vuejs.org/guide/migration/migration-build.html#known-limitations) and [preparations](https://v3.vuejs.org/guide/migration/migration-build.html#preparations). It is worth keeping in mind while performing the upgrade. And for large-scale/production-level applications, it makes a lot of sense to do it that way. It reduces the risk of change by having an [intermediary build](https://v3.vuejs.org/guide/migration/migration-build.html#installation).
+
+Perhaps for smaller ones like this one, though, there is another way. The Vue CLI is capable of generating a new Vue 3 project. If it is possible to copy the existing Vue components directly over to this newly-generated one, upgrading will be effortless. If the files do not copy for whatever reason, the intermediary build is a fallback plan.
+
+In a different directory, create the Vue 3 app via the Vue CLI using the same plugins as before: Unit-Testing (Jest) and TypeScript. In the original project, switch to a new branch. Delete the `/src` and `/tests` folders that came pre-loaded in the Vue 3 app and copy the Vue 2 corresponding directories over. Delete all files (except the .git and .gitignore) in the original project. Next, move the Vue 3 files over to the original project. Use the new branch to git compare changes between versions 2 and 3.
```

Comment:
```suggestion
In a different directory, create the Vue 3 app via the Vue CLI using the same plugins as before: Jest (for unit testing) and TypeScript. In the original project, switch to a new branch. Delete the `/src` and `/tests` folders that came pre-loaded in the Vue 3 app and copy the Vue 2 corresponding directories over. Delete all files (except the .git and .gitignore) in the original project. Next, move the Vue 3 files over to the original project. Use the new branch to git compare changes between versions 2 and 3.
```

Reply:
The Vue CLI prompts for `Unit-Testing` first, then you can select `Jest` following that, which is why I ordered it like that.

Reply:
Ahh makes sense.  Okay, scrap that lol

Reply:
updated